stdlib/LinearAlgebra/test/triangular.jl

```diff
@@ -651,4 +651,20 @@ end
     @test_throws ArgumentError LinearAlgebra.powm(A, 2.2)
 end
 
+# Issue 35058
+A = [
+       0.9999999999999998        4.649058915617843e-16        -1.3149405273715513e-16        9.9959579317056e-17
+       -8.326672684688674e-16        1.0000000000000004        2.9280733590254494e-16        -2.9993900031619594e-16
+       9.43689570931383e-16        -1.339206523454095e-15        1.0000000000000007        -8.550505126287743e-16
+       -6.245004513516506e-16        -2.0122792321330962e-16        1.183061278035052e-16        1.0000000000000002
+    ]
+B = [
+       0.09648289218436859        0.023497875751503007        0.0        0.0
+       0.023497875751503007        0.045787575150300804        0.0        0.0
+       0.0        0.0        0.0        0.0
+       0.0        0.0        0.0        0.0
+    ]
+C = sqrt(A*B*A')
+@test !any(isnan,C)
```

Comment:
Better
```jl
@test C^2 ≈ A*B*A'
```
which will also fail if there are NaN values in `C`.

Reply:
updated in 573d006